src/tasks.py

```diff
@@ -60,14 +62,23 @@ def get_task_sampler(
         "quadratic_regression": QuadraticRegression,
         "relu_2nn_regression": Relu2nnRegression,
         "decision_tree": DecisionTree,
+        "kalman_filter": KalmanFilter,
+        "noisy_kalman_filter": NoisyKalmanFilter
     }
+    print(curriculum)
     if task_name in task_names_to_classes:
         task_cls = task_names_to_classes[task_name]
         if num_tasks is not None:
             if pool_dict is not None:
                 raise ValueError("Either pool_dict or num_tasks should be None.")
-            pool_dict = task_cls.generate_pool_dict(n_dims, num_tasks, **kwargs)
-        return lambda **args: task_cls(n_dims, batch_size, pool_dict, **args, **kwargs)
+            #print(task_name == "kalman_filter")
+            if task_name == "kalman_filter":
+                #print(curriculum)
+                pool_dict = KalmanFilter.generate_pool_dict(n_dims, num_tasks, curriculum=curriculum, **kwargs)
+                return lambda **args: KalmanFilter(n_dims, batch_size, curriculum, pool_dict=pool_dict, **args, **kwargs)
+            else:    
+                pool_dict = task_cls.generate_pool_dict(n_dims, num_tasks, **kwargs)
+        return (lambda **args: KalmanFilter(n_dims, batch_size, curriculum, pool_dict, **args, **kwargs)) if task_name == "kalman_filter" else (lambda **args: task_cls(n_dims, batch_size, pool_dict, **args, **kwargs))
```

Comment:
You redo the check here in line 81 if task_name == "kalman_filter" which not necessary since if task_name was kalman_filter, line 78 would have returned already

Reply:
No. Consider the case where num_tasks is not None, and run through the get_task_sampler function again. The second if statement is needed